From 4e173cfdc04a73233e1ad0f4cb7020a57e109209 Mon Sep 17 00:00:00 2001
From: Pooventhiran G <pooventhiran.g@oss.qualcomm.com>
Date: Tue, 26 May 2026 12:14:25 +0530
Subject: [PATCH] net: skbuff: Add SKB-extension support to wireless drivers

Add SKB-extension support to wireless driver stack so that additional
data can be transported between wireless drivers and wireless stack.

Seamless Mobility Domain (SMD), part of P802.11bn/D1.4, is the first
wireless feature to use this extension.

Change-Id: I43fb2091d61046b1f7f3da78ebddbf5172babe3a
Signed-off-by: Pooventhiran G <pooventhiran.g@oss.qualcomm.com>
---
 include/linux/skbuff.h | 3 +++
 net/core/skbuff.c      | 6 ++++++
 2 files changed, 9 insertions(+)

diff --git a/include/linux/skbuff.h b/include/linux/skbuff.h
index fca8fcf9fcfd..b11aaa72aaac 100644
--- a/include/linux/skbuff.h
+++ b/include/linux/skbuff.h
@@ -4816,6 +4816,9 @@ enum skb_ext_id {
 #if IS_ENABLED(CONFIG_MCTP_FLOWS)
 	SKB_EXT_MCTP,
 #endif
+#if IS_ENABLED(CONFIG_WIRELESS)
+	SKB_EXT_WIRELESS,
+#endif
 #ifdef CONFIG_IPQ_PON
 	SKB_EXT_GEM,
 #endif
diff --git a/net/core/skbuff.c b/net/core/skbuff.c
index fb59a5365017..dc4896e424a3 100644
--- a/net/core/skbuff.c
+++ b/net/core/skbuff.c
@@ -88,6 +88,9 @@
 #ifdef CONFIG_ATHMEMDEBUG
 #include <linux/ath_memdebug.h>
 #endif
+#if IS_ENABLED(CONFIG_WIRELESS)
+#include <linux/skb_wireless.h>
+#endif
 
 #include "dev.h"
 #include "sock_destructor.h"
@@ -5178,6 +5181,9 @@ static const u8 skb_ext_type_len[] = {
 #ifdef CONFIG_IPQ_PON
 	[SKB_EXT_GEM] = SKB_EXT_CHUNKSIZEOF(struct gem_skb_ext),
 #endif
+#if IS_ENABLED(CONFIG_WIRELESS)
+	[SKB_EXT_WIRELESS] = SKB_EXT_CHUNKSIZEOF(struct wireless_skb_ext)
+#endif
 };
 
 static __always_inline unsigned int skb_ext_total_length(void)
-- 
2.34.1

